frontend/src/pages/task/[taskId]/edit.tsx

```diff
@@ -1,73 +1,78 @@
-import Button from "@/components/Button";
-import EmbeddedApp, { EmbeddedAppRef } from "@/components/EmbeddedApp";
+import { useTask, useTaskFile } from "@/api/collimator/hooks/tasks/useTask";
+import {
+  useUpdateTask,
+  useUpdateTaskFile,
+} from "@/api/collimator/hooks/tasks/useUpdateTask";
+import CrtNavigation from "@/components/CrtNavigation";
 import Header from "@/components/Header";
-import MaxScreenHeight from "@/components/layout/MaxScreenHeight";
-import { scratchAppHostName } from "@/utilities/constants";
-import { downloadBlob } from "@/utilities/download";
-import { readSingleFileFromDisk } from "@/utilities/file-from-disk";
+import MultiSwrContent from "@/components/MultiSwrContent";
+import PageHeader from "@/components/PageHeader";
+import TaskForm, { TaskFormValues } from "@/components/task/TaskForm";
 import { useRouter } from "next/router";
-import { useCallback, useMemo, useRef } from "react";
+import { useCallback } from "react";
+import { Container } from "react-bootstrap";
+import { defineMessages, FormattedMessage } from "react-intl";
+
+const messages = defineMessages({
+  submit: {
+    id: "EditTask.submit",
+    defaultMessage: "Save Task",
+  },
+});
 
 const EditTask = () => {
   const router = useRouter();
   const { taskId } = router.query as {
     taskId?: string;
   };
 
-  const iFrameSrc = useMemo(() => {
-    return `${scratchAppHostName}/edit/${taskId}`;
-  }, [taskId]);
-
-  const embeddedApp = useRef<EmbeddedAppRef | null>(null);
-
-  const onLoadTask = useCallback(async () => {
-    if (!embeddedApp.current) {
-      return;
-    }
-
-    // TODO: get the task from the API
-    // for now get it from the file system
+  const task = useTask(taskId);
+  const taskFile = useTaskFile(taskId);
+  const updateTask = useUpdateTask();
+  const updateTaskFile = useUpdateTaskFile();
 
-    const task = await readSingleFileFromDisk();
+  const onSubmit = useCallback(
+    async (formValues: TaskFormValues) => {
+      if (task.data && taskFile.data) {
+        if (formValues.blobChanged) {
+          await updateTaskFile(task.data.id, formValues.blob);
+        }
 
-    await embeddedApp.current.sendRequest({
-      procedure: "loadTask",
-      // typescript does not seem to notice that "arguments" is required but without the Omit<> it does, a bug?
-      arguments: task,
-    });
-  }, []);
+        await updateTask(task.data.id, {
+          title: formValues.title,
+          description: formValues.description,
+          type: formValues.type,
+        });
```

Comment:
We need to handle failures and display an appropriate message to the user.

In the future, we should handle this better to avoid a 2-phase commit while providing the same guarantees. If we centralize the blob storage (database-side), we could just create the blob in `updateTaskFile`, and reference it in the task when calling `updateTask` - making it the only operation actually changing the task.